client/src/editMovie.js

```diff
@@ -0,0 +1,139 @@
+import { React, useState } from "react";
+import MovieForm from "./form";
+import Button from "./button";
+
+var UpdatedGenres = ({ name }) => <li>{name}</li>;
+
+function EditMovie({
+  movieTitle,
+  moviePK,
+  currentGenres,
+  allGenres,
+  cancelClickHandler,
+  deleteClickHandler,
+}) {
+  var [checkedState, setCheckedState] = useState(initialCheck());
+  var [updatedTitle, setUpdatedTitle] = useState(movieTitle);
+  var [submitted, setSubmitted] = useState(false);
+  var [submittedGenreNames, setSubmittedGenreNames] = useState([currentGenres]);
+  var selectedGenrePKs = [];
+  //var selectedGenreNames = [];
+
+  /*Wrapper functions can be passed to child components, allowing the child component to update state in
+    its parent
+  */
+  var checkedStateWrapper = function (data) {
+    setCheckedState(data);
+  };
+  var submitStateWrapper = function (data) {
+    setSubmitted(data);
+  };
+  var updatedTitleWrapper = function (data) {
+    setUpdatedTitle(data);
+  };
+
+  /*Creates new array (initSelectedGenres) with "true" values for movie's current genres on initial rendering.
+  CurrentGenre's genre locations are checked with allGenres, to ensure that the genre's index in allGenres is
+  marked true in initSelectedGenres. This way, if the order of currentGenres changes, the correct genre will 
+  be associated with the movie when it is updated.
+
+  This function "checks" the checkboxes for the genres currently associated with the movie when user enters
+  the edit form.
+  */
+  function initialCheck() {
+    var initSelectedGenres = new Array(allGenres.length).fill(false);
+    currentGenres.forEach((movieGenre) => {
+      allGenres.forEach((genre, index) => {
+        if (genre.pk === movieGenre.pk) {
+          initSelectedGenres[index] = true;
+        }
+      });
+    });
+    return initSelectedGenres;
+  }
+
+  async function updateMovie() {

```

Comment:
In production, I would move a function like this up to app.js and pass a reference to it down through props or context.

editMovie doesn't need to know the details of how our fetch works, it just needs to be able to use it. By keeping all of the low level program details like the fetch URL up in app.js, there's only one place we need to look if we change it.